app/models/application_letter.rb

```diff
@@ -20,7 +20,23 @@ class ApplicationLetter < ActiveRecord::Base
   validates :grade, presence: true, numericality: { only_integer: true }
   validates :vegeterian, :vegan, :allergic, inclusion: { in: [true, false] }
   validates :vegeterian, :vegan, :allergic, exclusion: { in: [nil] }
+  validate :deadline_cannot_be_in_the_past, :if => Proc.new { |letter| !(letter.status_changed?) }
```

Comment:
You need a Proc or lambda:
http://stackoverflow.com/questions/6232099/rails-validation-if-proc-new-or-lambda
http://stackoverflow.com/questions/8848601/rails-how-to-validate-a-field-only-if-a-another-field-has-a-certain-value